website/content/configuration/_advanced_ipaddresspool_configuration.md

```diff
@@ -48,6 +48,52 @@ To specify a single IP address in a pool, use `/32` in the CIDR notation
 (e.g. `42.176.25.64/32`).
 {{% /notice %}}
 
+### Reduce scope of address allocation to specific Namespace and Service
+
+This option can be used to reduce the scope of particular IPAddressPool
+to set of namespaces and services, by adding an optional namespace/service
+selector.
+
+The use the pool with lowest value of priority in case of multiple matches.
+This is deterministic and will make the behaviour more clear to the users.
+
+A pool with no priority set will be used only if the pools with priority
+can't be used.
+
+Given a Service, if multiple matching IPAddressPools are available, MetalLB
+will check for the availability of IPs sorting the matching IPAddressPools
+by priority, starting from the highest to the lowest.
+If multiple IPAddressPools have the same priority, the choice will be
+random.
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: ippool-ns-service-alloc-sample
+  namespace: metallb-system
+spec:
+  addresses:
+    - 192.168.20.0/24
+  avoidBuggyIPs: true
+  serviceAllocation:
+    priority: 50
+    namespaces:
+      - namespace-a
+      - namespace-b
+    namespaceSelectors:
+      - matchLabels:
+          foo: bar
+    serviceSelectors:
+      - matchExpressions:
+          - {key: app, operator: In, values: [bar]}
+```
+
+The above IPAddressPool example is pinned to service(s) with has a label
+matching expression `key: app, operator: In, values: [bar]` and created
+either in `namespace-a` or `namespace-b` or any other namespace(s) which
+has a matching label `foo: bar`.
```

Comment:
I'd also add a note to say that if a service chooses a pool (or an ip belonging to a pool) which doesn't match the service will stay in pending.

Reply:
done.